packages/cli/src/ui/components/InputPrompt.tsx

```diff
@@ -210,37 +249,76 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
 
           // Insert @path reference at cursor position
           const insertText = `@${relativePath}`;
-          const currentText = buffer.text;
           const [row, col] = buffer.cursor;
 
           // Calculate offset from row/col
-          let offset = 0;
-          for (let i = 0; i < row; i++) {
-            offset += buffer.lines[i].length + 1; // +1 for newline
-          }
-          offset += col;
-
-          // Add spaces around the path if needed
-          let textToInsert = insertText;
-          const charBefore = offset > 0 ? currentText[offset - 1] : '';
-          const charAfter =
-            offset < currentText.length ? currentText[offset] : '';
+          const offset = logicalPosToOffset(buffer.lines, row, col);
 
-          if (charBefore && charBefore !== ' ' && charBefore !== '\n') {
-            textToInsert = ' ' + textToInsert;
-          }
-          if (!charAfter || (charAfter !== ' ' && charAfter !== '\n')) {
-            textToInsert = textToInsert + ' ';
-          }
-
-          // Insert at cursor position
-          buffer.replaceRangeByOffset(offset, offset, textToInsert);
+          // Insert at cursor position with auto spaces
+          insertAtOffsetWithAutoSpaces(offset, insertText);
         }
       }
     } catch (error) {
       console.error('Error handling clipboard image:', error);
     }
-  }, [buffer, config]);
+  }, [buffer, config, insertAtOffsetWithAutoSpaces]);
+
+  const expandPlaceholders = useCallback(
```

Comment:
as a follow up, please integrate this with the syntax highlighting for the input prompt. that will make it clear these placeholders are special and have been matched. This will make the case where the placeholder is edited less confusing as well as it will switch to plain text when it doesn't match.